File: boj/[1484] 다이어트/java/1484.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.List;

public class Main {

    public static void main(String[] args) throws IOException {
        final BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        final int G = Integer.parseInt(br.readLine());

        double start = 1.0;
        double end = 2.0;
        final List<Integer> answer = new ArrayList<>();
        while (start != end) {
            if ((Math.pow(end, 2.0) - Math.pow(start, 2.0)) > G) {
                start++;
            } else if ((Math.pow(end, 2.0) - Math.pow(start, 2.0)) == G) {
                answer.add((int) end);
                start++;
            } else {
                end++;
            }
        }

        if (answer.isEmpty()) {
            System.out.println(-1);
        } else {
            for (final int a : answer) {
                System.out.println(a);
            }
        }
    }
}

File: boj/[1484] 다이어트/kotlin/1484.kt
import kotlin.math.pow

fun main() {
    val G = readln().toInt()
    var start = 1.0
    var end = 2.0
    val answer = mutableListOf<Int>()

    while (start != end) {
        when {
            (end.pow(2) - start.pow(2)) > G.toDouble() -> start++
            (end.pow(2) - start.pow(2)) == G.toDouble() -> {
                answer.add(end.toInt())
                start++
            }
            else -> end++
        }
    }
    if (answer.isEmpty()) {
        println(-1)
    } else {
        answer.forEach { println(it) }
    }
}
